graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
```

Comment:
I used `utils.DigestReference` because that exact function is used inside the v2 pull functions to check if `loadManifest` should attempt to verify that the manifest digest matches or not - the check should be identical. This should be safe as the tag has already been parsed, so the only way it can contain a colon is if it has been specified by the digest format.

I dislike mixing tags and digests in the `tag` argument to Pull, but fixing that is a bigger refactoring effort and I do not know enough Go.